apps/eth/views.py

```diff
@@ -86,5 +88,51 @@ def get_queryset(self):
         LOG.debug('Getting tx details for a transaction hash.')
         queryset = self.queryset
         if settings.PROFILES_ENABLED:
-            queryset = queryset.filter(ethlistener__shipments__owner_id=self.request.user.id)
+            queryset = queryset.filter(Q(ethlistener__shipments__owner_id=self.request.user.id) |
+                                       Q(ethlistener__shipments__shipper_wallet_id=
+                                         self.request.query_params.get('wallet_id')) |
+                                       Q(ethlistener__shipments__moderator_wallet_id=
+                                         self.request.query_params.get('wallet_id')) |
+                                       Q(ethlistener__shipments__carrier_wallet_id=
+                                         self.request.query_params.get('wallet_id')))
```

Comment:
1. These `wallet_id` filters should not be applied if `wallet_id` is not specified (right now these filters will be applied on the `/api/v1/transactions/{transaction_hash}/` view, which doesn't require `wallet_id`). If `self.request.query_params.get('wallet_id')` returns `None`, as it does when the parameter is omitted, then this filter will include results with Shipments with a null moderator.
2. This case ^ should be added to the unit tests, and assert that the filters are working as intended.
3. Please add unit tests for the existing `/api/v1/transactions/{transaction_hash}/` endpoint.